pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -120,14 +127,15 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 	for _, hc := range allMachineHealthChecks.Items {
 		if hasMatchingLabels(&hc, machine) {
 			glog.V(4).Infof("Machine %s has a matching machineHealthCheck: %s", machineKey, hc.Name)
-			remediate(node)
+			return remediate(r, machine)
```

Comment:
We're deliberating ignoring the rest of `range allMachineHealthChecks.Items` - yes? if so, why don't the other items come into play?

Reply:
The initial approach is simple. A MachineHealthCheck CRD is not configurable, it just select machines based on labels, if there's a least one matching a machine it should be remediated.